integration-tests/src/test/java/tachyon/security/BlockWorkerClientAuthenticationIntegrationTest.java

```diff
@@ -76,43 +71,32 @@ public void simpleAuthenticationOpenCloseTest() throws Exception {
   @LocalTachyonClusterResource.Config(
       tachyonConfParams = {Constants.SECURITY_AUTHENTICATION_TYPE, "CUSTOM",
           Constants.SECURITY_AUTHENTICATION_CUSTOM_PROVIDER,
-          NameMatchAuthenticationProvider.FULL_CLASS_NAME},
-      startCluster = false)
+          NameMatchAuthenticationProvider.FULL_CLASS_NAME,
+          Constants.SECURITY_LOGIN_USERNAME, "tachyon"})
   public void customAuthenticationOpenCloseTest() throws Exception {
-    /**
-     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
-     * Tachyon Worker.
-     */
-    System.setProperty(Constants.SECURITY_LOGIN_USERNAME, "tachyon");
-    mLocalTachyonClusterResource.start();
     authenticationOperationTest();
   }
 
   @Test(timeout = 10000)
   @LocalTachyonClusterResource.Config(tachyonConfParams = {Constants.SECURITY_AUTHENTICATION_TYPE,
       "CUSTOM", Constants.SECURITY_AUTHENTICATION_CUSTOM_PROVIDER,
-      NameMatchAuthenticationProvider.FULL_CLASS_NAME}, startCluster = false)
+      NameMatchAuthenticationProvider.FULL_CLASS_NAME,
+      Constants.SECURITY_LOGIN_USERNAME, "tachyon"})
   public void customAuthenticationDenyConnectTest() throws Exception {
-    /**
-     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
-     * Tachyon Master during starting cluster.
-     */
-    System.setProperty(Constants.SECURITY_LOGIN_USERNAME, "tachyon");
-    mLocalTachyonClusterResource.start();
-
     // Using no-tachyon as loginUser to connect to Worker, the IOException will be thrown
-    System.setProperty(Constants.SECURITY_LOGIN_USERNAME, "no-tachyon");
-    // Clear the login user so that it will be reloaded and pick up our no-tachyon change
-    clearLoginUser();
+    ClientContext.getConf().set(Constants.SECURITY_LOGIN_USERNAME, "no-tachyon");
+
     mThrown.expect(IOException.class);
     mThrown.expectMessage("Failed to connect to the worker");
 
     BlockWorkerClient blockWorkerClient = new BlockWorkerClient(
         mLocalTachyonClusterResource.get().getWorkerAddress(),
-        mExecutorService, mLocalTachyonClusterResource.get().getWorkerTachyonConf(),
+        mExecutorService, ClientContext.getConf(),
         1 /* fake session id */, true, new ClientMetrics());
     try {
       Assert.assertFalse(blockWorkerClient.isConnected());
+      // Clear the login user so that it will be reloaded and pick up our no-tachyon change
+      clearLoginUser();
```

Comment:
`connect` implicitly invoke `LoginUser.get()` when client wants to connect to master. The `LoginUser` is a singleton. So if it is null, a login will be performed before `connect`. If not null, the value will be treated as current user.

The logic is, `connect` will fail because it first find `loginUser` cleared, so login with `no-tachyon` from conf, and then this user is rejected by server (the server is configured with an AuthenticationProvider which only allow user `tachyon`).

At the annotation above, user `tachyon` is set to start local cluster. And clean the user here so that `connect` could implicit login user `no-tachyon` and use it to connect.